app/templates/courses/courses-view.jade

```diff
@@ -115,23 +115,27 @@ mixin course-instance-body(courseInstance, classroom)
       - var arenaLevel = stats.levels.arena;
       - var projectLevel = stats.levels.project;
       if arenaLevel
-        - var arenaURL = "/play/ladder/"+arenaLevel.get('slug')+"/course/"+courseInstance.id;
+        - var levelSlug = view.originalSlugMap[arenaLevel.get('original')] || arenaLevel.get('slug');
+        - var arenaURL = "/play/ladder/"+levelSlug+"/course/"+courseInstance.id;
         a.play-btn.btn.btn-forest-alt.btn-lg.m-b-1(data-href=arenaURL, data-level-slug=arenaLevel.get('slug'), data-event-action="Students Play Arena")
           span(data-i18n="courses.play_arena")
       else if projectLevel
-        - var projectURL = "/play/level/"+projectLevel.get('slug')+"?course="+course.id+"&course-instance="+courseInstance.id;
+        - var levelSlug = view.originalSlugMap[projectLevel.get('original')] || projectLevel.get('slug');
+        - var projectURL = "/play/level/"+levelSlug+"?course="+course.id+"&course-instance="+courseInstance.id;
         a.play-btn.btn.btn-forest-alt.btn-lg.m-b-1(data-href=projectURL, data-level-slug=projectLevel.get('slug'), data-event-action="Students Play Project")
           span(data-i18n="courses.view_project")
       else
         a.btn.btn-default.btn-lg.m-b-1(disabled=true, data-i18n="courses.course_complete")
     else if stats.levels.next != stats.levels.first
       - var next = stats.levels.next;
-      - var levelURL = "/play/level/"+next.get('slug')+"?course="+courseInstance.get('courseID')+"&course-instance="+courseInstance.id;
+      - var levelSlug = view.originalSlugMap[next.get('original')] || next.get('slug');
+      - var levelURL = "/play/level/"+levelSlug+"?course="+courseInstance.get('courseID')+"&course-instance="+courseInstance.id;
       a.play-btn.btn.btn-forest.btn-lg.m-b-1(data-href=levelURL, data-level-slug=next.get('slug'), data-event-action="Students Continue Course")
         span(data-i18n="common.continue")
     else
       - var firstLevel = stats.levels.first;
-      - var levelURL = "/play/level/"+firstLevel.get('slug')+"?course="+courseInstance.get('courseID')+"&course-instance="+courseInstance.id;
+      - var levelSlug = view.originalSlugMap[firstLevel.get('original')] || firstLevel.get('slug');
+      - var levelURL = "/play/level/"+levelSlug+"?course="+courseInstance.get('courseID')+"&course-instance="+courseInstance.id;
```

Comment:
I was thinking about consolidating logic for building urls in a library:

https://github.com/codecombat/codecombat/blob/master/app/core/urls.coffee

And using it in templates like this:

https://github.com/codecombat/codecombat/blob/master/app/templates/courses/teacher-class-view.jade#L330

This would help minimize redundant logic, and make it easier to identify where in the app links are being generated by searching for the function. Could be used here as well, what do you think?